sdk/eventgrid/azure-eventgrid/azure/eventgrid/_helpers.py

```diff
@@ -84,7 +84,9 @@ def _get_authentication_policy(credential):
             credential=credential,
             name=constants.EVENTGRID_TOKEN_HEADER
         )
-    return authentication_policy
+    raise ValueError("The provided credential should be an instance of \
+        EventGridSharedAccessSignatureCredentialPolicy or AzureKeyCredentialPolicy"
```

Comment:
I guess before merging this PR we should use the azure-core type to avoid wasting time to do a PR message fix in a few days :)

Reply:
I agree with it :)